docs/modbus_set_error_user_data.md

```diff
@@ -0,0 +1,32 @@
+# modbus_set_error_user_data
+
+## Name
+
+modbus_set_error_user_data - set file stream to write log output
+
+## Synopsis
+
+```c
+void modbus_set_error_user_data(modbus_t *ctx, void* out_user_data);
```

Comment:
Are these really void* or should they explicitly be FILE * pointers?  The names and the examples and the descriptions don't really line up well here.

Reply:
this is form the original PR. This also was the "usual" way to pass arbitrary data for callback when I last did "C", so I didn't change it.